Added optional flag to respect absolute encoder positions

JosephBowkett opened the pull request:
Daniel, mind taking a look over this code to see if you agree with what it does?

EELS is about to test this on the robot, but it is meant to allow an actuator to ignore any value saved to disk and just read position from an absolute encoder at startup.

d-loret reviewed (commented):
Disclaimer: for the actuator driver, I'm not super familiar yet with the portion of the driver that reads the position from the pos files.

Having said that, if `fastcat::Actuator::SetOutputPosition` is the only function interacting with that, then it seems the code is doing the right thing.

My suggestion is to add a specific `ParseOptVal` overload for boolean values in the same fashion as [`ParseVal`](https://github.com/nasa-jpl/fastcat/blob/master/src/yaml_parser.cc#L134-L144).

d-loret commented on src/manager.cc:
```diff
@@ -919,8 +919,8 @@ bool fastcat::Manager::SetActuatorPositions()
 
-    MSG("Setting actuator: %s to saved pos: %lf", dev_name.c_str(),
-        find_pos_data->second.position);
+    //MSG("Setting actuator: %s to saved pos: %lf", dev_name.c_str(),
```
Why don't leaving the message in with the understanding that the code does nothing for absolute encoders.

JosephBowkett commented:
Huh good point on there already being a bool parsing option, I looked briefly but didn't see one. Yeah I'll make an optional bool version.

On removing the message I thought it might be a little confusing to see conflicting messages, given one says "setting actuator to..." and then afterwards it would say "ignoring actuator because absolute encoder". I'll see if there's a way to clear these up so it selectively prints them when using relative encoders. It also looks like there might need to be a change or two on the functional side of the code.

JosephBowkett commented:
Added the ParseOptVal for bool, and refactored the flag for absolute_encoder so that the manager.cc file knows which actuators to skip when reading and writing from/to the saved positions file.

JosephBowkett commented:
Tested on the 3 link EELS 1.0 robot, works as expected by only reading and writing to saved position file when an actuator does NOT have absolute_encoder: {True,true,TRUE} (other capitalizations like TrUE fail)

@d-loret, if the latest code looks logical to you, I'll merge this.
